geode-redis/src/main/java/org/apache/geode/redis/internal/RegionProvider.java

```diff
@@ -51,4 +67,24 @@ public RegionProvider(InternalCache cache) {
   public Region<String, Object> getConfigRegion() {
     return configRegion;
   }
+
+  /**
+   * Validates that the value passed in is a power of 2 and that it is not greater than
+   * {@link #REDIS_SLOTS}
+   *
+   * @throws ManagementException if there is a problem with the value
+   */
+  protected static void validateBucketCount(int buckets) {
+    if (buckets <= 0 || ((buckets & (buckets - 1)) != 0)) {
+      throw new ManagementException(
+          String.format(
+              "Could not start Redis Server - System property '%s' must be a power of 2. Configured value is invalid: %d",
```

Comment:
Technically, I don't know if we really need to restrict users to powers of two. I think your math will work fine with any number of buckets.

Reply:
If the slots count, fixed at 16k, is not evenly divisible by the bucket count then you end up with less slots in some buckets. If the data is evenly distributed across the buckets, as described in the Redis docs by use of CRC16, then data would be imbalanced too.

Reply:
I'll remove the restriction - it's going to be undocumented in any case (at least for now).